namer/namer.cc

```diff
@@ -629,12 +629,24 @@ class SymbolFinder {
         return foundDefs->addTypeMember(move(found));
     }
 
+    bool sendRecvIsT(const ast::Send &s) {
+        bool result = false;
+
+        typecase(
```

Comment:
Opted for a `typecase` here. Seems better than the hand-rolled `switch` alternative. Is it?

```cpp
        switch (receiver.tag()) {
            case ast::Tag::ConstantLit:
                return ast::cast_tree<ast::ConstantLit>(receiver)->symbol == core::Symbols::T();
            case ast::Tag::UnresolvedConstantLit:
                return ast::cast_tree<ast::UnresolvedConstantLit>(receiver)->cnst == core::Names::Constants::T();
            default:
                return false;
        }
```

Reply:
The compiler can convert typecase into code that is equivalent to the switch.  Pick whichever you think is more readable or your reviewer would like you to use. 😅

Reply:
I personally would have used

```cpp
if (auto *c = ast::cast_tree<ast::UnresolvedConstantLit>(s.recv)) {
  return c->cnst == core::Names::Constants::T();
} else if (auto *c = ast::cast_tree<ast::ConstantLit>(s.recv)) {
  return c->symbol == core::Symbols::T();
} else {
  return false;
}
```

Which _might not_ compile down to the same code as the type case / switch? But realistically all three are fine. Of the three I think my least favorite is the one that uses the `.tag()` explicitly, and leaving it with `typecase` is completely fine.

Reply:
I think I prefer `typecase`, though admittedly, I liked it more before the linter put `s.recv` on the same line as the first lambda 😅

Reply:
Looks like `typecase()` needs to be exhaustive. This PR is failing with a bunch of `Exception::raise(): not handled typecase case: ...` errors.

How do I express the equivalent of a `default:` case?